lib/lambda_ethereum_consensus/store/blob_db.ex

```diff
@@ -55,8 +72,55 @@ defmodule LambdaEthereumConsensus.Store.BlobDb do
     end
   end
 
+  @spec prune_blobs_older_than(non_neg_integer()) :: :ok | {:error, String.t()} | :not_found
+  def prune_blobs_older_than(slot) do
+    Logger.info("[BlobDb] Pruning started.", slot: slot)
+    last_finalized_key = slot |> block_root_key(0)
+
+    with {:ok, it} <- Db.iterate(),
+         {:ok, @block_root_prefix <> _, _value} <-
+           Exleveldb.iterator_move(it, last_finalized_key),
+         {:ok, keys_to_remove} <- get_block_root_keys_to_remove(it),
+         :ok <- Exleveldb.iterator_close(it) do
+      keys_to_remove |> Enum.each(&remove_blob_by_block_root_key/1)
+
+      Logger.info("[BlobDb] Pruning finished. #{length(keys_to_remove)} blobs removed.")
+    end
+  end
+
+  @spec get_block_root_keys_to_remove(list(binary()), :eleveldb.itr_ref()) ::
+          {:ok, list(binary())}
+  defp get_block_root_keys_to_remove(keys_to_remove \\ [], iterator) do
+    case Exleveldb.iterator_move(iterator, :prev) do
+      {:ok, <<@block_root_prefix, _rest::binary>> = block_root_key, _root} ->
+        [block_root_key | keys_to_remove] |> get_block_root_keys_to_remove(iterator)
+
+      _ ->
+        {:ok, keys_to_remove}
+    end
+  end
```

Comment:
Optional: maybe we can use `Enum.reduce_while` to get the key and delete it in the same iteration?

Reply:
Hmmm, let's not get stuck on this one too much, it's not even a performance improvement. We may think about it later on.

Reply:
Let me know if you have any additional comments on this PR or if we should approve it!